src/pages/product/product.js

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useState } from 'react'
+import { products } from '../../data/product.js'
+import { useCart } from '../../context/cart-context'
+import { Cateogery } from '../../data/cateogery'
+import {filterByLowHigh, filterByHighLow, getFilterByPrizeRangeVal, getFilterByRatings, getFilterByCateogry } from './utils'
+function Product() {
+    const [productsList, setProductList] = useState([])
+    const [cateogeryFilters, setCateogeryFilters] = useState([])
+    const [ratingsFilter, setRatingFilter] = useState(null)
+    const [prizeRangeFilter, setPrizeRangeFilter] = useState('High-Low')
+    const [rangeVal, setRangeVal] = useState(120000)
```

Comment:
use useReducer for multiple state management

Reply:
For practice purpose I have use usestate but in last my plan is to use usereducer.